src/GafferImage/Catalogue.cpp

```diff
@@ -1131,14 +1148,37 @@ void Catalogue::imageReceived( Gaffer::Plug *plug )
 	internalImage->driverClosed();
 }
 
+
+void Catalogue::plugSet( const Plug *plug )
+{
+	// Enforce that only one image may have a particular output index
+	auto currentImage = plug->parent<Image>();
+	if( currentImage && currentImage->parent() == imagesPlug() && plug == currentImage->outputIndexPlug() )
+	{
+		int newIndex = currentImage->outputIndexPlug()->getValue();
+		if( newIndex == 0 )
+		{
+			return;
+		}
+
+		for( const auto &image : Image::Range( *imagesPlug() ) )
+		{
+			if( image != currentImage && image->outputIndexPlug()->getValue() == newIndex )
+			{
+				image->outputIndexPlug()->setValue( 0 );
```

Comment:
This is going to break when you add unit tests for promoted plugs - it needs to set `image->outputIndexPlug()->source()`.